File: Backend/Backend/Models/Producto.cs
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
namespace Backend.Models;
public class Producto
{
    [Key]
    public int ProductoID { get; set; }
    public string Nombre { get; set; } = null!;
    public byte CategoriaID { get; set; }
    public decimal PrecioBase { get; set; }
    public string? UnidadMedida { get; set; }
    public bool EsCombo { get; set; }
    public DateTime FechaCreacion { get; set; }
    public bool Activo { get; set; }
    public List<ComboItem> ComboItems { get; set; } = new();
}

File: Flutter/lib/models/producto.dart
class Producto {
  final int productoID;
  final String nombre;
  final double precioBase;
  final int categoriaID;
  final String? unidadMedida;

  Producto({required this.productoID, required this.nombre, required this.precioBase, required this.categoriaID, this.unidadMedida});

  factory Producto.fromJson(Map<String, dynamic> json) => Producto(
    productoID: json['productoID'] ?? json['productoId'] ?? 0,
    nombre: json['nombre'] ?? '',
    precioBase: (json['precioBase'] ?? 0).toDouble(),
    categoriaID: json['categoriaID'] ?? json['categoriaId'] ?? 0,
    unidadMedida: json['unidadMedida']
  );
}
